src/Agent/NewRelic/Agent/Core/Agent.cs

```diff
@@ -439,14 +439,15 @@ public void RecordLogMessage(string frameworkName, object logEvent, Func<object,
 
                 var logMessage = getLogMessage(logEvent);
                 var logException = getLogException(logEvent);
+                var logContextData = _configurationService.Configuration.ContextDataEnabled ? getContextData(logEvent) : null;
 
-                // exit quickly if the message and exception are missing
-                if (string.IsNullOrWhiteSpace(logMessage) && logException is null)
+                // exit quickly if the message, exception and context data are missing
+                if (string.IsNullOrWhiteSpace(logMessage) && logException is null && (logContextData is null || logContextData.Count == 0))
```

Comment:
We need to ensure that the corresponding agent spec is updated to account for this change.
We should also ensure that this change is called out in the changelog.